src/Cache/CacheKeyAndTags.php

```diff
@@ -12,19 +13,19 @@ interface CacheKeyAndTags
      * @param  int|string|null $id
      * @param  array<string, mixed> $args
      */
-    public function key(?Authenticatable $user, bool $isPrivate, string $parentName, $id, string $fieldName, array $args): string;
+    public function key(?Authenticatable $user, bool $isPrivate, $id, array $args, ResolveInfo $resolveInfo): string;
 
     /**
      * Generate a tag for the parent.
      *
      * @param  int|string|null $id
      */
-    public function parentTag(string $parentName, $id): string;
+    public function parentTag($id, ResolveInfo $resolveInfo, ?string $parentName = null): string;
```

Comment:
Why keep `$parentName`? It is never passed.

Reply:
I see, maybe passing `ResolveInfo` is not such a great idea after all, since it pretty much forces them to be called only in resolvers. I can see use cases where keys or tags are generated somewhere else, so we might just go back to your original implementation of just adding `array $path` to `key()`.

Sorry for the back and forth, but I would rather go the extra mile when designing interfaces than to break public APIs often.

Reply:
Absolutely no problem. When it is a breaking change it is better to check all the possibilities.